modules/component/types/extractValueTypeFromAction.ts

```diff
@@ -0,0 +1,14 @@
+import {Action} from '@action-land/core'
+
+/**
+ * infers type of values from action set given action type's type
+ * @typeparam A set of actions
+ * @typeparam T type of action for which value type is to be inferred
```

Comment:
it can just say value, `value-type` is implied because you're describing type-lambdas anyway